craft_providers/bases/buildd.py

```diff
@@ -828,13 +816,19 @@ def _setup_wait_for_network(
 
         _check_deadline(deadline)
         while True:
-            proc = executor.execute_run(
-                ["getent", "hosts", "snapcraft.io"],
-                capture_output=True,
-                check=False,
-            )
-            if proc.returncode == 0:
-                return
+            try:
+                proc = executor.execute_run(
+                    ["getent", "hosts", "snapcraft.io"],
+                    check=False,
+                    timeout=1,
+                )
+            except subprocess.TimeoutExpired:
```

Comment:
I have a question about this:
> we seen too many times that [getent] "hangs"

Are you talking about `getent` hanging as in "stuck forever"?

I have been observing the issue where `getent` doesn't get stuck (it times itself out after 5 seconds), but the `while True` loop in this method runs forever and `getent` gets called again and again.  I believe this would be solved with using `deadline`.

I was trying to simulate the failure with charmcraft.  With or without this PR, the behavior is the same - charmcraft stays in the while loop forever with no user feedback (unless running with a high verbosity level).